Source/SwiftLintFramework/Rules/ClassDelegateProtocolRule.swift

```diff
@@ -79,6 +81,10 @@ public struct ClassDelegateProtocolRule: ASTRule, ConfigurationProviderRule {
     private func isClassProtocol(file: File, range: NSRange) -> Bool {
         return !file.match(pattern: "\\bclass\\b", with: [.keyword], range: range).isEmpty
     }
+    
+    private func isReferenceProtocol(file: File, range: NSRange) -> Bool {
+        return !file.match(pattern: "\\b(AnyObject|NSObjectProtocol)\\b", with: [.typeidentifier], range: range).isEmpty
```

Comment:
you should be able to just use `dictionary.inheritedTypes` instead, as it already contains all inherited types.